src/main/java/com/javaschool/mobile/controller/ContractController.java

```diff
@@ -0,0 +1,66 @@
+package com.javaschool.mobile.controller;
+
+import com.javaschool.mobile.entity.Contract;
+import com.javaschool.mobile.entity.Tariff;
+import com.javaschool.mobile.service.ContractService;
+import com.javaschool.mobile.service.OptionService;
+import com.javaschool.mobile.service.TariffService;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Controller
+@RequestMapping("/admin")
+public class ContractController {
+
+    private final ContractService contractService;
+
+    private final TariffService tariffService;
+
+    private final OptionService optionService;
+
+    public ContractController(ContractService contractService, TariffService tariffService, OptionService optionService) {
+        this.contractService = contractService;
+        this.tariffService = tariffService;
+        this.optionService = optionService;
+    }
+
+    @GetMapping("/contracts/{phone}")
```

Comment:
{phone} should be in query param

Reply:
Like this?

    @RequestMapping("/contracts/phoneNumber")
    public String findContractByPhoneNumber(@RequestParam("phone") Long phone, Model model){
        var contract = contractService.findContractByPhoneNumber(phone);
        var options = contract.getOptions();
        model.addAttribute("contract",contract);
        model.addAttribute("options",options);
        return "contract";